src/transformers/pipelines/base.py

```diff
@@ -75,7 +75,7 @@ def _pad(items, key, padding_value, padding_side):
         # Others include `attention_mask` etc...
         shape = items[0][key].shape
         dim = len(shape)
-        if dim == 4:
+        if key == "pixel_values":
```

Comment:
`pixel_values` is a consistent key used across `transformers` that doesn't require padding so I think it's ok to update that condition to cover ImagePGT

Reply:
How does that solve the problem since currently the [feature extractor of ImageGPT is outputting pixel values](https://github.com/huggingface/transformers/blob/daf520b03345e270fa8cca81c4dbc3d9f46b413c/src/transformers/models/imagegpt/feature_extraction_imagegpt.py#L174)?

Reply:
The key part is this : `        # flatten to (batch_size, height*width)`

So the tensor is `dim == 2` even though it does represent `pixel_values`. 
Since the `dim==4` was intended to catch images and afaik there are no other image like tensor aside from `pixel_values` I am switching condition to catch the `dim==2` tensor too.

Images don't get padding basically, the feature extractor is responsible for making them of the same size so they can be stacked

Reply:
But ImageGPT using `pixel_values` as a key is a mistake fixed [here](https://github.com/huggingface/transformers/pull/16872) so this test won't catch `ImageGPT` anymore.

Reply:
I see, then will the feature extractor provide a  `padding_value`?   Would it make sense in that context? 

If yes, then we don't need to modify anything here, it will pad nicely automatically.
If no, this means that padding (hence batching at inference) doesn't really mean anything and we shouldn't rely on it, no ? Or are the `input_ids` produces always of the same `seq_len` (in which case we can go for something like @ydshieh proposed which is to pad only when tensors stack even if  `padding_value` does not exist.)

Reply:
I think the processed images will all be of the same size.